### PR TITLE
Add test deps to deploy action

### PR DESCRIPTION
**Pull request by IAlibay**
This _should_ work. It'll attempt to run `pip install ${test_deps}` before running the tests.

### CODE REVIEW

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -31,6 +31,7 @@ jobs:
           test_submission: true
           package_name: MDAnalysisData
           module_name: 'MDAnalysisData'
+          test_deps: 'pytest pytest-mock'
```
Comment (orbeckst) [inline .github/workflows/deploy.yaml:34]: Note: This information is effectively duplicated from the information in setup.py.

**Review by orbeckst — APPROVED** [review]
I am not super-thrilled that we are tracking the test dependency here and in setup.py's `tests_require`. However, `tests_require` is deprecated in setuptools anyway (see https://setuptools.pypa.io/en/latest/deprecated/changed_keywords.html ) so we will need to update all of that eventually (perhaps only `pyproject.toml` without deprecated junk?) and I also don't have another better immediate solution so I'll happily merge.

Thanks for the quick fix, @IAlibay !

File: .github/workflows/deploy.yaml (hunk 1 above)
Comment (IAlibay) [reply .github/workflows/deploy.yaml:34]: Unfortunately how this is stored in setup.py isn't consistent, even across our own projects - lack of enforced guidelines makes these things necessary.

A good example of this is `doc` vs `docs` (which has driven me crazy lately).

Comment (orbeckst) [reply .github/workflows/deploy.yaml:34]: I believe you!

**Pull request merged**